db/migrate/20221129162916_add_user_ref_to_posts.rb

```diff
@@ -0,0 +1,5 @@
+class AddUserRefToPosts < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :posts, :users, null: false, foreign_key: true
```

Comment:
- [ ] Kindly change user to author so as to generate author_id i.e `t.references :author, foreign_key: { to_table: 'users' } `